python/test/regression/test_cast_matmul.py

```diff
@@ -0,0 +1,233 @@
+"""
```

Comment:
I actually suggest adding it as a simple function in https://github.com/openai/triton/blob/main/python/test/regression/test_functional_regressions.py